force-app/main/default/lwc/miscPermissionBasedUI/__tests__/miscPermissionBasedUI.test.js

```diff
@@ -19,53 +19,122 @@ describe('c-misc-permission-based-u-i', () => {
         while (document.body.firstChild) {
             document.body.removeChild(document.body.firstChild);
         }
+        jest.clearAllMocks();
     });
 
-    it('displays the correct UI when custom permission is true', () => {
+    // Helper function to wait until the microtask queue is empty. This is needed for promise
+    // timing when calling imperative Apex.
+    async function flushPromises() {
+        return Promise.resolve();
+    }
+
+    it('displays the correct UI when custom permission is true', async () => {
+        mockPermission.mockReturnValueOnce({
+            ...mockModule,
+            default: true
+        });
+
+        let MiscPermissionBasedUI;
```

Comment:
Maybe we can move this segment of code to beforeEach() method as it is being used in all of the tests.